dali/pipeline/operators/reader/loader/loader.h

```diff
@@ -167,14 +168,19 @@ class Loader {
   virtual Index Size() = 0;
 
  protected:
+  virtual void MoveToNextShard(Index current_index) {
+    if (IsNextShard(current_index)) {
+      Reset(!stick_to_shard_);
+    }
+  }
   // Reset reader to the first sample
-  virtual void Reset() = 0;
+  virtual void Reset(bool move_to_beginning) = 0;
```

Comment:
You still can move to the beginning of shard or whole set, so maybe:
```suggestion
  virtual void Reset(bool wrap_to_shard) = 0;
```
(this would reverse the booleans with which we call it).

Reply:
Maybe we could provide the default `Reset()` here as it seems to be the same among some of the readers.

Reply:
Common reset won't work, code is the same but variable that it resets differs:
- current_index_
- current_frame_idx_
- current_sequence_

Reply:
Changed `move_to_beginning` -> `wrap_to_shard`.